### ISSUE TITLE
Explore a lightweight notification UI to display confirmation feedback for actions to users

### ISSUE THREAD

**Issue by joyceerhl** [microsoft/vscode#166891]
Motivating scenarios:
- Remote Repositories has a use case for displaying a one-off message on save to let the user know that their uncommitted changes will be saved in browser storage (similar to the Figma save toast). 
- Another use case is to provide confirmation that actions like copying a GitHub permalink to clipboard succeeded. 

Since this could get visually noisy IMO there should be a global toggle for this so that users can opt out.

**Comment by joyceerhl** [comment]
Settings sync displays a toast after enablement succeeds which could also benefit from a lightweight treatment:
<img width="467" alt="image" src="https://user-images.githubusercontent.com/30305945/204611255-7053eaed-8ec4-4481-b8ec-c6b9f6117e5c.png">

**Comment by heartacker** [comment]
https://github.com/microsoft/vscode/issues/153693

**Comment by daviddossett** [comment]
There are at least two obvious approaches to this. More ideas welcome:

## Adding a new toast API

This a lightweight UI component that features simple text + 1-2 actions. It has a set timeout and no other UI affordances (e.g. explicit close, expand/collapse, etc.). One advantage here is to able to separate ephemeral messages from formal persistent notifications. Today we often use notifications for temporary states like progress (signing into GH) and confirmation (settings sync is turned on). This would enable us to use notifications only for important messages/actions that need to be kept in the notifications center (errors, extension recommendations, etc.).

https://user-images.githubusercontent.com/25163139/211642317-e718e094-808c-4eed-9891-b75cfe313fe3.mp4

Some other examples...

<img width="785" alt="CleanShot 2023-01-10 at 11 36 38@2x" src="https://user-images.githubusercontent.com/25163139/211645388-25f35745-9968-4f60-9a3d-de4b7b6adbf1.png">


## Extending notification API with `showConfirmationMessage` type

This would sit alongside the existing info, warning, and error notification types. The key differences here could be that they disappear quickly and aren't persisted in the notification center—which could be messy/confusing.

https://user-images.githubusercontent.com/25163139/211642564-e23c30ff-259f-4a37-9563-9dea100597cc.mp4

**Comment by bpasero** [comment]
I think another location for notification feedback that is somewhere global but not where our current notifications are is a bit confusing because then we have 2 locations to look at and not one.

Could the lightweight notification be shown close to where the interaction happened and then disappear automatically?

**Comment by daviddossett** [comment]
@bpasero Are you thinking of it still like a toast but somehow anchored to the element that was clicked? 

One challenge there is that not everything will be from a button click. The remote repositories example in the issue description is on such example.

Here's some prior art with similar decoupling of notifications and toasts:

## Figma

Figma uses toasts for things like basic UI feedback as well as reminding you about certain functionality. E.g. when you hit cmd + S it tells you "Figma auto-saves your work" as a nice little reminder. 

https://user-images.githubusercontent.com/25163139/211882897-37688d14-d355-4788-bf86-262c60d5b9cf.mp4

## Outlook 

Outlook uses toasts for similar types of UI feedback. Their notifications are also handled in a separate area similar to Figma.

https://user-images.githubusercontent.com/25163139/211882187-4a5cc4e8-a9b4-4ccc-8414-30be28a6e6a5.mp4

**Comment by daviddossett** [comment]
Here's what it could look like to use a sort of anchored tooltip element:

![CleanShot 2023-01-11 at 10 49 30](https://user-images.githubusercontent.com/25163139/211892815-2fdf0e16-a958-4c42-aae8-7e4680d80b39.gif)

![CleanShot 2023-01-11 at 10 48 56](https://user-images.githubusercontent.com/25163139/211892846-002330b4-7b5d-4dc7-a9c3-a97deffb7197.gif)

**Comment by bpasero** [comment]
Yeah I really like the notifications in context. To me the global locations are always a fallback because we did not find a way to show notifications in context.

**Comment by daviddossett** [comment]
Most of the feedback from the team was positive for the tooltip style. I think it would make sense to reuse our actual tooltips since we have so much positioning logic already baked in. 

<img width="243" alt="CleanShot 2023-01-11 at 15 54 08@2x" src="https://user-images.githubusercontent.com/25163139/211942986-b81c1469-b672-4894-8e74-c86f0f9b18a7.png">

I don't know for sure if there is any sort of timeout logic, however. It may also be nice to have support for icons/progress too.

**Comment by siegenthalerroger** [comment]
Preserving context really is king and should be done if there is a context to bind to. If there isn't any context and it really is global, I would suggest not having a new, additional, location for these on top of the two already existing ones (bottom bar and bottom right notifications).

**Comment by joyceerhl** [comment]
The MakeCode extension exposes a `Create MakeCode Share Link` action which could potentially benefit from a confirmation feedback treatment. Today the extension displays confirmation by printing a message to the output channel which it focuses after the link is generated and copied. cc @jwunderl
